### PR TITLE
improvement(sct.py): Add junit argus client call

### LINKED ISSUE #7252: Operator tests should advertise xunit/junit results
So we can see them in Jenkins (and Argus?) as they are running regular pytest or whatnot test suite that does provide that.
Currently, it looks like everything case failed, which is far from true.
@vponomaryov and I were looking at this earlier this week.

### PR DESCRIPTION
**Pull request by k0machi**
This change updates the Junit collect stage to submit the resulting xml
file to argus, allowing Argus to store JUnit results

Fixes #7252

### Testing

- [x] [Jenkins](https://jenkins.scylladb.com/job/scylla-staging/job/alexey/job/functional-k8s-local-kind-aws-junit-test-test/2)
- [x] [Argus](https://argus.scylladb.com/test/afbd4be7-a69c-4b1c-bb04-3aaba5e99771/runs?additionalRuns[]=6a12936c-4da7-45fc-83d0-41500e964093)

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by k0machi** [comment]
Waiting on #7433 to be merged first as it contains hydra image required for this PR to work.

File: sct.py [hunk 1]
```diff
@@ -1161,8 +1161,8 @@ def run_pytest(target, backend, config, logdir):
         sys.exit(1)
     return_code = pytest.main(['-s', '-v', f'--junit-xml={junit_file}', '-p', 'no:warnings', target])
     test_config = get_test_config()
-    # temp, until we'll have API for sending it
-    test_config.argus_client().submit_sct_logs([LogLink(log_name=junit_file.name, log_link=str(junit_file))])
+    with open(junit_file, "rt", encoding="utf-8") as junit_xml:
```
Comment (fruch) [inline sct.py:1164]: try using Path APIs to read the file

File: sct.py [hunk 2]
```diff
@@ -1161,8 +1161,8 @@ def run_pytest(target, backend, config, logdir):
...
+        test_config.argus_client().sct_submit_junit_report(file_name=junit_file.name, raw_content=junit_xml.read())
```
Comment (soyacz) [inline sct.py:1165]: how big these files are?
What is the limit of payload?

Comment (k0machi) [reply sct.py:1165]: No limit, they are saved as blobs into a separate table. The files are pretty small, about 50kb or so

File: sct.py (hunk 1 above)
Comment (k0machi) [reply sct.py:1164]: Okay

**Comment by k0machi** [comment]
Double checking with another job, but seems to be ready (merge #7433 first)

**Review by soyacz — APPROVED** [review]
LGTM

**Comment by k0machi** [comment]
Looks good, @fruch

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**